shared/app/src/commonMain/kotlin/com/adammcneilly/pocketleague/shared/app/eventdetail/EventDetailPresenter.kt

```diff
@@ -24,10 +35,53 @@ class EventDetailPresenter(
             mutableStateOf(EventDetailDisplayModel.placeholder)
         }
 
+        var selectedStageId by remember {
+            mutableStateOf(displayModel.getStageSummaries().first().stageId)
+        }
+
+        var matchesForSelectedStage by remember {
+            mutableStateOf(
+                listOf(
+                    MatchDetailDisplayModel.placeholder,
+                    MatchDetailDisplayModel.placeholder,
+                    MatchDetailDisplayModel.placeholder,
+                ),
+            )
+        }
+
+        LaunchedEffect(Unit) {
+            eventRepository
+                .getEvent(eventId)
+                .map(Event::toDetailDisplayModel)
+                .onEach { eventDetail ->
+                    displayModel = eventDetail
+                }
+                .launchIn(this)
+
+            snapshotFlow {
+                selectedStageId
+            }.flatMapLatest { stageId ->
+                matchRepository.getMatchesForEventStage(
```

Comment:
We may not want to fetch matches every time the selected stage changes, perhaps we can keep a map of matches by stage ID and only fetch them once when the user first scrolls to it?